src/utils/map-controls.js

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// EVENT HANDLING PARAMETERS
+const PITCH_MOUSE_THRESHOLD = 5;
+const PITCH_ACCEL = 1.2;
+const ZOOM_ACCEL = 0.01;
+
+const SUBSCRIBED_EVENTS = [
+  'panstart',
+  'pan',
+  'panend',
+  'pinchstart',
+  'pinch',
+  'pinchend',
+  'doubletap',
+  'wheel'
+];
+
+export default class MapControls {
+  /**
+   * @classdesc
+   * A class that handles events and updates mercator style viewport parameters
+   */
+  constructor() {
+    this.events = SUBSCRIBED_EVENTS;
+  }
+
+  /**
+   * Update the state of the control
+   * @param {MapState} state.mapState - current map state
+   * @param {Function} state.onChangeViewport - callback
+   */
+  setState({mapState, onChangeViewport, isDragging, perspectiveEnabled}) {
+    this.mapState = mapState;
+    this.onChangeViewport = onChangeViewport;
+    this.isDragging = isDragging;
+    this.perspectiveEnabled = perspectiveEnabled;
+  }
+
+  /**
+   * Callback for events
+   * @param {hammer.Event} event
+   */
+  handle(event) {
```

Comment:
There is something here that gives off an odd vibe. This class is clearly written around handling hammer specific events, and yet it doesn't handle the registration of those events.  So this knowledge of the event manager becomes spread out between the components. Perhaps it is the right call, but it seems like an important thing to make clear.

Again if MapControls is not exported, I think we could land this now and take another look at it before we make the class public.